exasol_advanced_analytics_framework/interface/create_event_handler_udf.py

```diff
@@ -1,10 +1,162 @@
+from exasol_bucketfs_utils_python.bucketfs_factory import BucketFSFactory
+from exasol_bucketfs_utils_python.bucketfs_location import BucketFSLocation
+from exasol_data_science_utils_python.preprocessing.sql.schema.schema_name import \
+    SchemaName
+from exasol_data_science_utils_python.preprocessing.sql.schema.table_name import \
+    TableName
+
+from exasol_advanced_analytics_framework.event_handler.event_handler_base \
+    import EventHandlerBase
+from exasol_advanced_analytics_framework.event_handler.event_handler_context \
+    import EventHandlerContext
+from exasol_advanced_analytics_framework.event_handler.event_handler_result \
+    import EventHandlerReturnQuery, EventHandlerResultBase
+from exasol_advanced_analytics_framework.event_handler.event_handler_state \
+    import EventHandlerState
+from exasol_advanced_analytics_framework.context_wrapper.udf_context_wrapper \
+    import UDFContextWrapper
+
 
 class CreateEventHandlerUDF:
+
     def __init__(self, exa):
         self.exa = exa
 
     def run(self, ctx) -> None:
-        ctx.emit("return query")
-        ctx.emit("status")
-        ctx.emit("query 1")
-        ctx.emit("query 2")
+        # get and set method parameters
+        iter_num = ctx[0]  # iter_num
+        event_handler_module = ctx[1]  # event_handler_module
+        event_handler_class = ctx[2]  # event_handler_class_name
+        bucketfs_connection = ctx[3]  # bucketfs_connection_name
+        parameters = ctx[4]  # event_handler_parameters
+
+        bucketfs_location = BucketFSFactory().create_bucketfs_location(
+            url=bucketfs_connection.address,
+            user=bucketfs_connection.user,
+            pwd=bucketfs_connection.password,
+            base_path=parameters["base"]
+        )
+        latest_bucketfs_path = PurePosixPath(
+            f"{event_handler_class}_{str(iter_num)}.pkl")
+
+        # load the latest (create if not) event handler state object
+        latest_state = self._load_latest_state(
+            iter_num, event_handler_class, event_handler_module, parameters,
+            bucketfs_location, latest_bucketfs_path)
+        event_handler_context: EventHandlerContext = latest_state.context
+        event_handler: EventHandlerBase = latest_state.event_handler
+
+        # call the user code
+        udf_context = self._create_udf_context_wrapper(ctx)
+        result: EventHandlerResultBase = event_handler.handle_event(
+            udf_context, event_handler_context)
+
+        # save the current state
+        iter_num += 1
+        current_bucketfs_path = PurePosixPath(
+            f"{event_handler_class}_{str(iter_num)}.pkl")
+        current_state = EventHandlerState(event_handler_context, event_handler)
+        self._save_current_state(
+            bucketfs_location, current_bucketfs_path, current_state)
+
+        # remove previous state
+        self._remove_previous_state(
+            bucketfs_location, latest_bucketfs_path)
```

Comment:
refactor: extract to method

Reply:
I didn't understand the refactoring you mean

Reply:
Extract the storing new state and removing old state into a method.